### PR TITLE
test: introduce a script to test flag `--comm`

### LINKED ISSUE #212: Add an e2e test to verify the `--comm` option works fine
**Is your feature request related to a problem? Please describe.**
Add an e2e test to verify the `--comm` (filter by process name) option works fine

**Describe the solution you'd like**
1. Add a bash script file in `testdata/` which contains the main test logic. (like this one: `test_docker_filter_by_pid.sh`), including two cases:
    1. target process start , then kyanos start , expected kyanos can watch  the target process's network.
    2. kyanos start, then target process start,  expected kyanos can watch  the target process's network.

3. Add a step in workflow : .github/workflows/test.yml which executes the test script file in 1.

### PR DESCRIPTION
**Pull request by spencercjh**
CLOSE #212

### CODE REVIEW

File: testdata/test_filter_by_comm.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+. $(dirname "$0")/common.sh
+set -ex
```
Comment (ccoVeille) [inline testdata/test_filter_by_comm.sh:3]: Why not adding -o pipefail?

Comment (spencercjh) [reply testdata/test_filter_by_comm.sh:3]: I copied these scaffold codes from other steps. I'm not sure what the side effects of changing the set option would be.

**Comment by hengyoush** [comment]
HTTPS may require sending more than two requests over a long connection to be correctly captured 😂.   
Can this test be simplified by using curl to request any website (via http), and then using `--comm curl`? Finally check, when grepping, filter for the domain name (e.g., if curl requests baidu, grep for baidu.com). 

What do you think? @spencercjh

File: testdata/test_filter_by_comm.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    cat "${BEFORE_LNAME}" | grep "127.0.0.1:4443"
+
+    # server start after kyanos
+    timeout 40 ${CMD} watch --debug-output http --comm python3 2>&1 | tee "${AFTER_LNAME}" &
+    timeout 30 python3 ./testdata/start_https_server.py  &
+    sleep 10
+    curl --insecure https://127.0.0.1:4443 &>/dev/null || true
+    wait
+
+    cat "${AFTER_LNAME}" | grep "127.0.0.1:4443"
```
Comment (hengyoush) [inline testdata/test_filter_by_comm.sh:31]: Filtering by ip:port alone may not be accurate, as only the final request and response being printed can be considered successful. I think we can filter by some request info like domain.

File: testdata/test_filter_by_comm.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+BEFORE_LNAME="${FILE_PREFIX}_filter_by_comm_before.log"
+AFTER_LNAME="${FILE_PREFIX}_filter_by_comm_after.log"
+
+function test_filter_by_comm() {
+    openssl req -x509 -newkey rsa:2048 -keyout server.pem -out server.pem -days 365 -nodes -subj "/C=US/ST=California/L=San Francisco/O=My Company/CN=localhost"
+
+    pip install --break-system-packages ssl || true
+
+    # server start before kyanos
+    timeout 40 python3 ./testdata/start_https_server.py  &
```
Comment (hengyoush) [inline testdata/test_filter_by_comm.sh:16]: Sorry, using HTTP might be more appropriate. I misled you.

File: testdata/test_filter_by_comm.sh (hunk 1 above)
Comment (hengyoush) [reply testdata/test_filter_by_comm.sh:3]: I think we can use this `set -euxo pipefail`. It should be no side effects .

**Comment by spencercjh** [comment]
> HTTPS may require sending more than two requests over a long connection to be correctly captured 😂.   
> 
> Can this test be simplified by using curl to request any website (via http), and then using `--comm curl`? Finally check, when grepping, filter for the domain name (e.g., if curl requests baidu, grep for baidu.com). 
> 
> 
> 
> What do you think? @spencercjh 

OK 👌

File: testdata/test_filter_by_comm.sh [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+FILE_PREFIX="/tmp/kyanos"
+BEFORE_LNAME="${FILE_PREFIX}_filter_by_comm_before.log"
+AFTER_LNAME="${FILE_PREFIX}_filter_by_comm_after.log"
+
+function test_filter_by_comm() {
+    # server start before kyanos
+    timeout 40 python3 ./testdata/start_http_server.py  &
+    timeout 30 ${CMD} watch --debug-output http --comm python3 2>&1 | tee "${BEFORE_LNAME}" &
+    sleep 10
+    curl http://127.0.0.1:8080 &>/dev/null || true
```
Comment (hengyoush) [inline testdata/test_filter_by_comm.sh:15]: You can use testdata/https-request/http_request.go to request the server in a long connection mode. Currently, the short connection is not very stable and still needs to be optimized.

File: testdata/test_filter_by_comm.sh [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+    cat "${BEFORE_LNAME}"
+    cat "${BEFORE_LNAME}" | grep "Host: 127.0.0.1:8080" | grep "\\[side\\]=server"
+}
+
+function test_filter_by_client_comm() {
+    # client start after kyanos
+    timeout 40 ${CMD} watch --debug-output http --comm https-request 2>&1 | tee "${AFTER_LNAME}" &
+    sleep 10
+    timeout 30 ./testdata/https-request/https-request 'http://github.com' 40 &
```
Comment (hengyoush) [inline testdata/test_filter_by_comm.sh:27]: use " http://ipinfo.io" to test, response of github.com  is too big.

**Comment by spencercjh** [comment]
> We could use " http://ipinfo.io" to test, because response of github.com is too big.

It's amazing, the test doesn't report the occasional error after I changed it as you said. Why is that?

**Comment by hengyoush** [comment]
> > We could use " http://ipinfo.io" to test, because response of github.com is too big.
> 
> It's amazing, the test doesn't report the occasional error after I changed it as you said. Why is that?

May be some bug 😨, huge response may cause some error, i need to take a closer look. Thank you for spotting the problem!

**Comment by hengyoush** [comment]
It seems to have failed again, there should be a bug. I may need to analyze the case of the client side again. You can comment out the case of the client side and only run the case of the server side.😂

**Review by hengyoush — APPROVED** [review]
LGTM.

**Pull request merged**